src/compiler/checker.ts

```diff
@@ -23411,7 +23411,6 @@ namespace ts {
             const assumeInitialized = isParameter || isAlias || isOuterVariable || isSpreadDestructuringAssignmentTarget || isModuleExports || isBindingElement(declaration) ||
                 type !== autoType && type !== autoArrayType && (!strictNullChecks || (type.flags & (TypeFlags.AnyOrUnknown | TypeFlags.Void)) !== 0 ||
                 isInTypeQuery(node) || node.parent.kind === SyntaxKind.ExportSpecifier) ||
-                node.parent.kind === SyntaxKind.NonNullExpression ||

```

Comment:
@sandersn Would you help me understand why we don't allow initialization of a type if it's (well in AST it's its parent) `NonNullExpress`?

Reply:
I think I understand why, cause if the type is already determined then we should assumed that it's initialized.